code/modules/hydroponics/grown/misc.dm

```diff
@@ -244,3 +244,283 @@
+		playsound(user, W.hitsound, 50, 1, -1)
+		icon_state = "coconut_carved"
+		desc = "A coconut. This one's got a hole in it."
+		name = "coconut"
+		defused = TRUE
+		fused = FALSE
+		fusedactive = FALSE
+		if(!seed.get_gene(/datum/plant_gene/trait/glow))
+			set_light(0, 0.0)
+		return
+	//IGNITING NADE LOGIC
+	if(fusedactive == FALSE && fused == TRUE)
+		var/lighting_text = W.ignition_effect(src, user)
+		if(lighting_text)
+			user.visible_message("<span class='warning'>[user] ignites [src]'s fuse!</span>", "<span class='userdanger'>You ignite the [src]'s fuse!</span>")
+			fusedactive = TRUE
+			defused = FALSE
+			playsound(src, 'sound/effects/fuse.ogg', 100, 0)
+			message_admins("[ADMIN_LOOKUPFLW(user)] ignited a coconut bomb for detonation at [ADMIN_VERBOSEJMP(user)] "+ pretty_string_from_reagent_list(reagents.reagent_list))
+			log_game("[key_name(user)] primed a coconut grenade for detonation at [AREACOORD(user)].")
+			addtimer(CALLBACK(src, .proc/prime), 5 SECONDS)
+			icon_state = "coconut_grenade_active"
+			desc = "RUN!"
+			if(!seed.get_gene(/datum/plant_gene/trait/glow))
+				light_color = "#FFCC66" //for the fuse
+				set_light(3, 0.8)
+			return
+
+	//ADDING A FUSE, NADE LOGIC
+	if (istype(W,/obj/item/stack/sheet/cloth) || istype(W,/obj/item/stack/sheet/durathread))
+		if (carved == TRUE && straw == FALSE && fused == FALSE)
+			user.show_message("<span class='notice'>You add a fuse to the coconut!</span>", 1)
+			W.use(1)
+			fused = TRUE
+			icon_state = "coconut_grenade"
+			desc = "A makeshift bomb made out of a coconut. You estimate the fuse is long enough for 5 seconds."
+			name = "coconut bomb"
+			return
+	//ADDING STRAW LOGIC
+	if (istype(W,/obj/item/stack/sheet/mineral/bamboo) && opened == TRUE && straw == FALSE && fused == FALSE)
+		user.show_message("<span class='notice'>You add a bamboo straw to the coconut!</span>", 1)
+		straw = TRUE
+		W.use(1)
+		icon_state += "_straw"
+		desc = "You can already feel like you're on a tropical vacation."
+	//OPENING THE NUT LOGIC
+	if (carved == FALSE && chopped == FALSE)
+		if(W.tool_behaviour == TOOL_SCREWDRIVER)
+			user.show_message("<span class='notice'>You make a hole in the coconut!</span>", 1)
+			carved = TRUE
+			opened = TRUE
+			reagent_flags = OPENCONTAINER
+			ENABLE_BITFIELD(reagents.reagents_holder_flags, OPENCONTAINER)
+			icon_state = "coconut_carved"
+			desc = "A coconut. This one's got a hole in it."
+			playsound(user, W.hitsound, 50, 1, -1)
+			return
+		else if(W.sharpness)
+			user.show_message("<span class='notice'>You slice the coconut open!</span>", 1)
+			chopped = TRUE
+			opened = TRUE
+			reagent_flags = OPENCONTAINER
+			ENABLE_BITFIELD(reagents.reagents_holder_flags, OPENCONTAINER)
+			spillable = TRUE
+			icon_state = "coconut_chopped"
+			desc = "A coconut. This one's sliced open, with all its delicious contents for your eyes to savour."
+			playsound(user, W.hitsound, 50, 1, -1)
+			return
```

Comment:
These are pretty much the same except for the icon_state, message and description, you could more or less merge them and use a new "screwed" variable for the aforementioned variables and message.